api/v1alpha1/bundleinstance_types.go

```diff
@@ -29,41 +29,55 @@ const (
 	TypeHasValidBundle       = "HasValidBundle"
 	TypeInvalidBundleContent = "InvalidBundleContent"
 	TypeInstalled            = "Installed"
-
-	ReasonBundleLookupFailed       = "BundleLookupFailed"
-	ReasonBundleLoadFailed         = "BundleLoadFailed"
-	ReasonReadingContentFailed     = "ReadingContentFailed"
-	ReasonErrorGettingClient       = "ErrorGettingClient"
-	ReasonErrorGettingReleaseState = "ErrorGettingReleaseState"
-	ReasonInstallFailed            = "InstallFailed"
-	ReasonUpgradeFailed            = "UpgradeFailed"
-	ReasonReconcileFailed          = "ReconcileFailed"
-	ReasonCreateDynamicWatchFailed = "CreateDynamicWatchFailed"
-	ReasonInstallationSucceeded    = "InstallationSucceeded"
+	TypeProgressing          = "Progressing"
+
+	ReasonBundleLookupFailed         = "BundleLookupFailed"
+	ReasonBundleLoadFailed           = "BundleLoadFailed"
+	ReasonReadingContentFailed       = "ReadingContentFailed"
+	ReasonErrorGettingClient         = "ErrorGettingClient"
+	ReasonErrorGettingReleaseState   = "ErrorGettingReleaseState"
+	ReasonInstallFailed              = "InstallFailed"
+	ReasonUpgradeFailed              = "UpgradeFailed"
+	ReasonReconcileFailed            = "ReconcileFailed"
+	ReasonCreateDynamicWatchFailed   = "CreateDynamicWatchFailed"
+	ReasonInstallationSucceeded      = "InstallationSucceeded"
+	ReasonMaxGeneratedBundlesReached = "MaxGenerationReached"
+	ReasonCleanupFailed              = "CleanupFailed"
 )
 
 // BundleInstanceSpec defines the desired state of BundleInstance
 type BundleInstanceSpec struct {
 	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
 	ProvisionerClassName string `json:"provisionerClassName"`
+	// Label selector for Bundles resources.
+	Selector *metav1.LabelSelector `json:"selector"`
+	// Template describes the generated Bundle that this instance will manage.
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +kubebuilder:validation:Schemaless
+	Template *BundleTemplate `json:"template"`
+}
```

Comment:
Can we make these properties be a bit more explicit? By giving them `Bundle` prefixes it makes this a bit more obvious what these fields are used for. I'm not entirely devoted to this exact fix but maybe we can be clearer here. Plus, it matches the struct names.
```suggestion
	// Label selector for Bundles resources.
	BundleSelector *metav1.LabelSelector `json:"selector"`
	// BundleTemplate describes the generated Bundle that this instance will manage.
	// +kubebuilder:pruning:PreserveUnknownFields
	// +kubebuilder:validation:Schemaless
	BundleTemplate *BundleTemplate `json:"template"`
}
```

Reply:
I'd like to keep the changes here minimal given the size of this PR already :laughing:. That said: I don't have a strong opinion here as these changes would only affect the Go types as users would still be typing `template` and `selector` as that's the json key. Maybe we can revisit this over time?

Reply:
Sure, we can revisit at another time. I will say that my original point was more to edit that `json` values as well. I'll update that diff and unresolve so you can see this comment but feel free to close it right after. I think it is fine to edit later since we're still in iteration station.

Reply:
Ah okay that makes more sense. In that case, that's where I'd start pushing back as this PR is implementing the design that had outlined those fields. We can revisit that concern over time.